src/dist/replication/common/fs_manager.h

```diff
@@ -90,19 +105,52 @@ class fs_manager
     bool for_each_dir_node(const std::function<bool(const dir_node &)> &func) const;
     void update_disk_stat();
 
+    // when executing "disk rebalance", we need find min and max node(sorted_by_density)
+    static bool sorted_by_density(const std::shared_ptr<dir_node> left,
```

Comment:
```suggestion
    static bool sorted_by_density(const std::shared_ptr<dir_node>& left, const std::shared_ptr<dir_node>& right)
```
use reference '&'

Reply:
Oh, levy5307 suggest it's not necessary for share_ptr.

Reply:
he's wrong. Copying shared_ptr increases the internal atomic counter. It's not free.

Reply:
Yeah, I have modified it